src/main/kotlin/org/rust/ide/annotator/RsErrorAnnotator.kt

```diff
@@ -244,22 +244,22 @@ class RsErrorAnnotator : Annotator, HighlightRangeExtension {
     private fun checkImpl(holder: AnnotationHolder, impl: RsImplItem) {
         val trait = impl.traitRef?.trait ?: return
         val traitName = trait.name ?: return
-
-        val canImplement = trait.functionList.associateBy { it.name }
-        val mustImplement = canImplement.filterValues { it.isAbstract }
+        
```

Comment:
We now resolve the trait twice: here and in the `toImplementOverride`. We could just return `null` from `toImplementOverride` if it's not resolved and return from here (see a comment for `toImplementOverride`). Here, we could only check that the trait reference exists and skip the further analysis if it doesn't: `if (impl.traitRef == null) return`